R/tools-def-auto.R

```diff
@@ -88,7 +88,9 @@ create_tool_def <- function(topic,
     cli::cli_rule(cli::style_bold("Response"))
   }
 
-  chat <- chat_openai(system_prompt = tool_prompt, model = model, echo = echo)
+  if(!inherits(chat,"Chat"))
+    chat <- chat_openai(system_prompt = get_tool_prompt(), model = model, echo = echo)
```

Comment:
A more common pattern would be

```R
chat <- chat %||% chat_openai(system_prompt = get_tool_prompt(), model = model, echo = echo)
```

I also think it would be easier if we just used the chat as a template, and overrode the existing turns and system prompt, e.g.

```R
chat$clone()
chat$set_system_prompt(get_tool_prompt())
chat$set_turns(list())
```